lib/smart_city/dataset.ex

```diff
@@ -103,11 +103,27 @@ defmodule SmartCity.Dataset do
         technical: Technical.new(tech)
       })
 
-    {:ok, struct}
+    errors = return_errors(struct)
+
+    case errors do
+      [] -> {:ok, struct}
+      _ -> {:error, errors}
+    end
```

Comment:
No need to keep a copy of the errors, if all we're going to do is evaluate them in a `case`.

```elixir
case return_errors(struct) do
  [] -> {:ok, struct}
  errors -> {:error, errors}
end
```